### PR TITLE
More progress on initial feedback. This pull request includes the

### PR DESCRIPTION
**Pull request by ncdkim**
following updates:

Consolidation of nfp_cdev, nfpcmd_dev, and nfp_dev structures
Replace TO_LE16/32_MEM FROM_LE16/32_MEM calls with native ones
Removed dependency on modversions.h
Removed some unused variables
Use native smp_mb__before_atomic instead of NF wrapper
Use kzalloc instead of kmalloc+memset
Indentation fix in nfp_fops

### CODE REVIEW

File: drivers/staging/ncipher/hostif.c [hunk 1]
```diff
@@ -1122,14 +1120,13 @@ static int nfp_release(struct inode *node, struct file *file)
...
-open : nfp_open,
-release : nfp_release,
+static const struct file_operations nfp_fops = { owner: THIS_MODULE,
+	poll : nfp_poll,
+	write : nfp_write,
+	read : nfp_read,
+	unlocked_ioctl : nfp_unlocked_ioctl,
+	open : nfp_open,
+	release : nfp_release,
 };
```
Comment (jessorensen) [inline drivers/staging/ncipher/hostif.c]: I'd say we should use C99 .foo = style initalizers here.

File: drivers/staging/ncipher/osif.c [hunk 2]
```diff
@@ -35,90 +35,78 @@ int nfp_config_inl(struct nfp_cdev *pdev, int offset, unsigned int *res)
...
 {
 	int oserr;
-	long task_state;
 
-	task_state = current->state;
-	COPY_TO_USER(ubuf, kbuf, len, oserr);
-	current->state = task_state;
+	oserr = copy_to_user(ubuf, kbuf, len) ? -EFAULT : 0;
 	return nfp_error(oserr);
 }
```
Comment (jessorensen) [inline drivers/staging/ncipher/osif.c]: These are still pretty much dear wrappers, so we should get rid of them completely.

File: drivers/staging/ncipher/hostif.c (hunk 1 above)
Comment (ncdkim) [reply drivers/staging/ncipher/hostif.c]: Done.

File: drivers/staging/ncipher/osif.c (hunk 2 above)
Comment (ncdkim) [reply drivers/staging/ncipher/osif.c]: Ok, we've gotten rid of them now.

**Comment by ncdkim** [comment]
We've also completed work on streamlining osif.h and I'm going to close this PR and make a new one. The comments you've made in this PR have been completed and I'll highlight them in the new PR.

**Pull request closed without merging**